scripts/build_codelabs

```diff
@@ -32,24 +34,126 @@ function is_xcpretty_available() {
   fi
 }
 
+# Check that flags have arguments.
+#
+# Exits if the flag does not have an argument.
+function check_arguments() {
+  if [ -z $1 ]; then
+    echo "Argument must be provided to flag."
+    exit 1
+  fi
+}
+
+# Checks that the argument is a valid reference.
+#
+# Exits if the reference is not valid.
+function check_reference() {
+  reference=$(git ls-remote https://github.com/material-components/material-components-ios.git $1 | wc -l)
+  if [ $reference -eq 0 ]; then
+    echo "Invalid reference name."
+    exit 1
+  fi
+}
+
+# Checks that the argument is a valid commit.
+#
+# Exits if the commit is not valid.
+function check_commit() {
+  if [ ! $(git cat-file commit $1) ]; then
+    echo "Invalid commit SHA."
+    exit 1
+  fi
+}
+
 # Parse command-line arguments.
 #
 # Note that we're following the command-line exit status convention of zero
 # to mean "success."
 verbose=1
-for i in "$@"; do
-  case $i in
+all_codelabs=0
+has_source=0
+
+# Set default branch to this branch
+export DEFAULT_BRANCH=$(git symbolic-ref --short HEAD)
+
+while [ $# -gt 0 ]; do
+  case $1 in
     -v|--verbose)
       verbose=0
       shift
       ;;
+    -rb|--remote-branch)
+      shift
+      check_arguments $1
+      check_reference $1
+      if [ $has_source -eq 1 ]; then
+        echo "Only one of --remote-branch/--commit/--tag allowed."
+        exit -1
+      fi
+      export TEST_BRANCH=$1
+      has_source=1
+      shift
+      ;;
+    -c|--commit)
+      shift
+      check_arguments $1
+      check_commit $1
+      if [ $has_source -eq 1 ];then
+        echo "Only one of --remote-branch/--commit/--tag allowed."
+        exit -1
+      fi
+      export TEST_COMMIT=$1
+      has_source=1
+      shift
+      ;;
+    -t|--tag)
+      shift
+      check_arguments $1
+      check_reference $1
+      if [ $has_source -eq 1 ];then
+        echo "Only one of --remote-branch/--commit/--tag allowed."
+        exit -1
+      fi
+      export TEST_TAG=$1
+      has_source=1
+      shift
+      ;;
+    -a|--all)
+      all_codelabs=1
+      shift
+      ;;
+    -h|--help)
+      echo
+      echo "Builds MDC iOS Codelabs."
+      echo
+      echo "The default branch of MDC iOS that is used for testing is your current branch."
```

Comment:
Consider:
By default the codelabs are tested against the remote branch tracking your current branch.